dev/com.ibm.ws.threading/src/com/ibm/ws/threading/PolicyExecutor.java

```diff
@@ -167,20 +167,56 @@
 
     /**
      * Registers a one-time callback to be invoked asynchronously
+     * when the count of running tasks exceeds the specified maximum.
+     * If a concurrency callback is already registered, this replaces
+     * the previous registration.
+     * To unregister an existing callback without replacing,
+     * specify a null value for the callback.
+     * The callback is automatically unregistered upon shutdown.
+     *
+     * @param max threshold for maximum concurrency beyond which the callback should be notified.
+     * @param callback the callback, or null to unregister.
+     * @return callback that was replaced or removed by the new registration.
+     *         null if no previous callback was in place.
+     * @throws IllegalStateException if the executor has been shut down.
+     */
+    Runnable registerConcurrencyCallback(int max, Runnable callback);
+
+    /**
+     * Registers a one-time callback to be invoked asynchronously
+     * the difference between task start time and submit time exceeds the specified maximum delay.
+     * If a late start callback is already registered, this replaces
+     * the previous registration.
+     * To unregister an existing callback without replacing,
+     * specify a null value for the callback.
+     * The callback is automatically unregistered upon shutdown.
+     *
+     * @param max threshold for maximum concurrency beyond which the callback should be notified.
```

Comment:
"@param max threshold for maximum concurrency beyond which the callback should be notified." - This line should be for maxDelay.

Reply:
again, good catch, will update